alfi/solver.py

```diff
@@ -629,6 +674,65 @@ def function_space(self, mesh, k):
         Q = FunctionSpace(mesh, elep)
         return MixedFunctionSpace([V, Q])
 
+    def get_transfers(self):
+        V = self.Z.sub(0)
+        Q = self.Z.sub(1)
+        if self.stabilisation_type in ["burman", None]:
+            qinject = NullTransfer().inject
+        elif self.stabilisation_type in ["gls", "supg"]:
+            if self.lag_stabilisation: # we don't need to inject pressure then
+                qinject = NullTransfer().inject
+            else:
+                qinject = inject
+        else:
+            raise ValueError("Unknown stabilisation")
+        if self.hierarchy == "bary":
+            vtransfer = SVSchoeberlTransfer((self.nu, self.gamma), self.tdim, self.hierarchy)
+            self.vtransfer = vtransfer
+            transfers = {
+                V.ufl_element(): (vtransfer.prolong, vtransfer.restrict if self.restriction else restrict, inject),
+                Q.ufl_element(): (prolong, restrict, qinject)
+            }
+        else:
+            transfers = {
+                Q.ufl_element(): (prolong, restrict, qinject)
+            }
+        return transfers
+
+    def configure_patch_solver(self, opts):
+        patchlu3d = "mkl_pardiso" if self.use_mkl else "umfpack"
+        patchlu2d = "petsc"
+        opts["patch_pc_patch_sub_mat_type"] = "seqaij"
+        opts["patch_sub_pc_factor_mat_solver_type"] = patchlu3d if self.tdim > 2 else patchlu2d
+
+    def distribution_parameters(self):
+        return {"partition": True, "overlap_type": (DistributedMeshOverlapType.VERTEX, 2)}
+
+
+class TaylorHoodSolver(NavierStokesSolver):
```

Comment:
I guess this is identical to the ScottVogeliusSolver except you override the `function_space` method?

Reply:
Yes, the `residual` method looks the same as `ScottVogeliusSolver`?